lib/xcodeproj/scheme/abstract_scheme_action.rb

```diff
@@ -51,29 +56,37 @@ def pre_actions=(pre_actions)
       end
 
       # @param [ExecutionAction] pre_action
-      #        The ExecutionAction to add to the PreActions node associated with this action
+      #        Add an action to the list of actions to run before this scheme action.
+      #        It can be either a 'Run Script' or a 'Send Email' action.
       #
       def add_pre_action(pre_action)
+        return if pre_and_post_actions_disabled
```

Comment:
doing this check here seems weird -- why is it not valid to access the pre / post actions, even if Xcode won't run them?

Reply:
I was trying to support only what has a purpose in Xcode, I wanted to avoid putting things in the `.xcscheme` file that are not gonna be used and maybe one day cause a weird Xcode crash? Maybe I'm being overly precautious though?

Reply:
Thoughts here? Maybe I'm missing something 🤔

Reply:
I think we should still allow accessing and modifying these attributes, regardless of whether they're disabled. It's up to the caller to determine if they want behavior conditional on things being enabled or not

Reply:
Done!